packages/core/src/extensions/DraggableBlocks/DraggableBlocksPlugin.ts

```diff
@@ -217,52 +222,131 @@ function dragStart(e: DragEvent, view: EditorView) {
+  function addBlock(coords: { left: number; top: number }) {
+    blockButtonsFrozen = true;
+
+    const pos = options.editor.view.posAtCoords(coords);
+    if (!pos) {
+      return;
+    }
+
+    const blockInfo = getBlockInfoFromPos(options.editor.state.doc, pos.pos);
+    if (blockInfo === undefined) {
+      return;
+    }
 
-  let menuShown = false;
-  let addClicked = false;
+    const { contentNode, endPos } = blockInfo;
+
+    // Creates a new block if current one is not empty for the suggestion menu to open in.
+    if (contentNode.textContent.length !== 0) {
+      const newBlockInsertionPos = endPos + 1;
+      const newBlockContentPos = newBlockInsertionPos + 2;
+
+      options.editor
+        .chain()
+        .BNCreateBlock(newBlockInsertionPos)
+        .BNSetContentType(newBlockContentPos, "textContent")
+        .setTextSelection(newBlockContentPos)
+        .run();
+    }
+
+    // Focuses and activates the suggestion menu.
+    options.editor.view.focus();
+    options.editor.view.dispatch(
+      options.editor.view.state.tr
+        .scrollIntoView()
+        .setMeta(SlashMenuPluginKey, {
+          // TODO import suggestion plugin key
+          activate: true,
+          type: "drag",
+        })
+    );
+  }
 
-  const onShow = () => {
-    menuShown = true;
+  function deleteBlock(coords: { left: number; top: number }) {
+    dragHandleMenu.hide();
+
+    const pos = options.editor.view.posAtCoords(coords);
+    if (!pos) {
+      return;
+    }
+
+    options.editor.commands.BNDeleteBlock(pos.pos);
+  }
+
+  // Initializes params for use in drag handle, drag handle menu, and add block button factories.
+  const addBlockButtonParams: AddBlockButtonParams = {
+    addBlock: () => addBlock({ left: 0, top: 0 }),
+    blockBoundingBox: new DOMRect(),
   };
-  const onHide = () => {
-    menuShown = false;
+  const dragHandleParams: DragHandleParams = {
+    blockBoundingBox: new DOMRect(),
   };
-  const onAddClicked = () => {
-    addClicked = true;
+  const dragHandleMenuParams: DragHandleMenuParams = {
+    deleteBlock: () => deleteBlock({ left: 0, top: 0 }),
+    dragHandleBoundingBox: new DOMRect(),
   };
 
+  // Creates drag handle, drag handle menu, and add block button editor elements.
+  const addBlockButton = options.addBlockButtonFactory(addBlockButtonParams);
+  const dragHandle = options.dragHandleFactory(dragHandleParams);
+  const dragHandleMenu = options.dragHandleMenuFactory(dragHandleMenuParams);
+
+  // Declares additional listeners to attach to the drag handle element for drag/drop & menu opening.
+  const dragStartCallback = (event: DragEvent) =>
+    dragStart(event, options.editor.view);
+  const dragEndCallback = (_event: DragEvent) => unsetDragImage();
+  const clickCallback = (_event: MouseEvent) => {
+    dragHandleMenu.show(dragHandleMenuParams);
+    blockButtonsFrozen = true;
+  };
+
+  function addDragHandleListeners() {
```

Comment:
I would remove the dependency on `element` here, but pass `dragStartCallback` and similar handlers to the factory. It's then the responsibility of the implementation (e.g.: blocknote/react) to attach the event listeners and call the callbacks